js/test/test.js

```diff
@@ -107,34 +107,60 @@ if (settings && settings.skip) {
     process.exit ();
 }
 
-//-----------------------------------------------------------------------------
+
+// ### common language specific methods ###
+async function runTesterMethod(exchange, methodName, ... args) {
+    return await tests[methodName](exchange, ... args);
+}
+
+function testMethodAvailableForCurrentLang(methodName) {
+    return methodName in tests && tests[methodName] !== undefined;
+}
+
+function findValueIndexInArray (arr, value) {
+    return arr.indexOf (value);
+}
+
+function exceptionMessage (exc) {
+    return '[' + exc.constructor.name + '] ' + exc.message.slice (0, 200);
+}
+
+// ### end of language specific common methods ###
+
+// ----------------------------------------------------------------------------
+// ### AUTO-TRANSPILER-START ###
+// ----------------------------------------------------------------------------
 
 async function test (methodName, exchange, ... args) {
-    console.log ('Testing', exchange.id, methodName, '(', ... args, ')');
+    // We don't need to print-out the skipped tests in terminal, as it doesn't add any value, because we already know the unsupported methods from exchange-capabilities
     if (exchange.has[methodName]) {
-        return await (tests[methodName] (exchange, ... args));
+        if (testMethodAvailableForCurrentLang(methodName)) {
+            console.log ('Testing', exchange.id, methodName, '(', ... args, ')');
+            return await runTesterMethod(exchange, methodName, ... args);
+        }
     }
 }
 
 async function testSymbol (exchange, symbol) {
-
     await test ('loadMarkets', exchange);
     await test ('fetchCurrencies', exchange);
     await test ('fetchTicker', exchange, symbol);
     await test ('fetchTickers', exchange, symbol);
     await test ('fetchOHLCV', exchange, symbol);
     await test ('fetchTrades', exchange, symbol);
-
-    if (exchange.id === 'coinbase') {
-
-        // nothing for now
-
-    } else {
-
-        await test ('fetchOrderBook', exchange, symbol);
-        await test ('fetchL2OrderBook', exchange, symbol);
-        await test ('fetchOrderBooks', exchange);
-    }
+    await test ('fetchOrderBook', exchange, symbol);
+    await test ('fetchL2OrderBook', exchange, symbol);
+    await test ('fetchOrderBooks', exchange);
+    await test ('fetchBidsAsks', exchange);
+    await test ('fetchTradingLimits', exchange);
+    await test ('fetchFundingRates', exchange);
+    await test ('fetchFundingRate', exchange, symbol);
+    await test ('fetchFundingRateHistory', exchange, symbol);
+    await test ('fetchIndexOHLCV', exchange, symbol);
+    await test ('fetchMarkOHLCV', exchange, symbol);
+    await test ('fetchPremiumIndexOHLCV', exchange, symbol);
+    await test ('fetchStatus', exchange);
+    await test ('fetchTime', exchange);
```

Comment:
above I list all tests for all unified methods. however, inside `test` method if it doesn't find the existing test file for any of them, they will be skipped.